### DISCUSSION TITLE
Add package.json > biome config support

### DISCUSSION THREAD

**Discussion by chneau** [biomejs/biome#2062] [Ideas]
Prettier configuration can be set in package.json instead of creating "yet" another file in the root of the project.
I believe in VSCode, when you have the prettier extension, you get the autocompletion on package.json>prettier.

Would it be nice to have the same UX for biome?

**Comment by SuperchupuDev** [comment]
does biome even look at package.json for configuration?

**Reply by chneau** [reply]
Woops, I forgot I was on the vscode repo, my bad

**Reply by chneau** [reply]
Opened the discussion on the main repo https://github.com/biomejs/biome/discussions/2061

**Comment by arendjr** [comment]
Biome does have the ability to load `package.json` now, but I’m not sure about the desirability of adding Biome’s own configuration to it as well. Our config loading is already very complex, and a feature like this would complicate matters for monorepos, for instance, which may have a single `biome.json` but multiple `package.json` files spread all over.

**Comment by SuperchupuDev** [comment]
personally, i don't think cluttering package.json with third party configs is a good practice nor something biome should encourage. when uploading a package to npm, the file is uploaded as-is which can make the package size slightly bigger. this is probably not an issue at a small scale, but think about how many gigabytes of bandwidth and watts have been wasted on *just* downloading and storing unnecessary package.json fields for all downloads of packages that happen to clutter package.json that could be avoided using separate config files

**Comment by ematipico** [comment]
I don't think I will ever agree to have this as an option:
- Our configuration is very big, and we risk to make `package.json` even bigger.
- If you use it in a `package.json` that you publish on npmjs.org, you end up sharing this configuration with other users, and they won't even use it.
- You can't use the schema for auto-completion.

**Reply by Conaclos** [reply]
I second this.
Moreover, using `package.json` as a central place for configuration is now highly discouraged.
More and more tools are giving up their support of a such practice.
